### DISCUSSION TITLE
Error fungsi select(), unused argument

### DISCUSSION THREAD

**Discussion by tu849035** [teamalgoritma/community#380] [Q&A] [answered]
Salam.

Muncul error seperti tampilan di bawah saat saya menggunakan fungsi `select()` dari dplyr.
![image](https://user-images.githubusercontent.com/107348211/193438213-cb8b0072-ea45-4d2a-9c7d-866bc76ee514.png)
Padahal sebelumnya tidak error, namun ketika saya run ulang, jadi error.
Saya coba cek materi rmd inclass yg sebelumnya juga jadi error, padahal waktu inclass tidak ada error.
![image](https://user-images.githubusercontent.com/107348211/193438284-7a973331-327e-4479-a209-733bd8589c8b.png)

Mohon bantuan solusi & sarannya.

Terima Kasih.

**Comment by onlyphantom** [comment, marked as answer]
In the first code chunk, you didn't run line 61 yet (it looks like you've executed line 62 to 64 only), so the line that loads `dplyr` has not yet execute. 

Because of this, your `select` is not the one exported from `dplyr`. You can make sure you run the code chunk again with the library executed. Alternatively, you can also just explicitly call it:

```
wholesale <-  wholesale %>% 
    mutate(Channel = as.factor(channel)) %>% 
    dplyr::select(-Region)
```

**Reply by tu849035** [reply]
Greetings.

Actually I already put all the library I need in one chunk and I have run it all before. But the error occur.
![image](https://user-images.githubusercontent.com/107348211/193448570-f1f0f79d-4adc-4d19-97bd-5db00e21dcbd.png)
It doesn't have error minutes bbefor. But, when I try to re-run the all the chunks again, the error occur. Even in the inclass rmd file, there is no error before. And then now it has an error.
This happen only for `dplyr` other library work normally.

wholesale <-  wholesale %>% 
    mutate(Channel = as.factor(channel)) %>% 
    dplyr::select(-Region)

But that syntax can solve the problem.

Thank You.

**Reply by onlyphantom** [reply]
OK, for clarification sake, you loaded `dbplyr` on line 31, and that too has a `select` function, and so this is a collision and developers call this "namespace pollution". When you do `select`, because `dbplyr` is loaded last, that `select` overwrites the one imported at line 25. 

Actually `tidyverse` alone brings in `dplyr` as well, so this select function is overloaded three (3!!) times bro. First line 23, then line 25, then line 31. I recommend spending some time actually cutting down the redundancies :) 

1. Take away dependencies you don't need, and if you only use it sparingly, use the `::` syntax instead. 
2. Some of the packages don't need to be loaded, for example `ggplot2` and `tidyr` and `dplyr` can all be removed because they are brought in on line 23 by `tidyverse`. Alternatively, remove `tidyverse`. https://www.tidyverse.org/packages/

This will make you work cleaner, less redundant, load faster, and when you deploy using shiny or something else in the future, it saves memory and cost. :)

Good luck!

**Reply by tu849035** [reply]
OMG. It's my fault.
I thought I have to load all the library for every different function I want to use.

Thank You so much. It helps a lot.
